src/lib/util/FileHandler.ts

```diff
@@ -94,4 +94,18 @@ export class FileHandler {
 			});
 		});
 	}
+
+	tmpDirWithCleanup(): Promise<string> {
+		return new Promise<string>((res, rej) => {
+			// Ask tmp to gracefully clean up everything on process exit.
+			tmp.setGracefulCleanup();
+			return tmp.dir({unsafeCleanup: true}, (err, name) => {
```

Comment:
At what point does the tmp directory clean up get triggered?

Reply:
It will get cleaned up when the process exits.